crates/e2e/tests/e2e/replace_order.rs

```diff
@@ -156,6 +172,36 @@ async fn try_replace_executed_order_test(web3: Web3) {
         .unwrap();
     let expected_body = String::from_utf8(expected_body_bytes.to_vec()).unwrap();
     assert_eq!(error_message, expected_body);
+
+    // Continue automining so our order can be executed
+    web3.api::<TestNodeApi<_>>()
+        .enable_automine()
+        .await
+        .expect("Must be able to disable auto-mining");
+
+    tracing::info!("Waiting for the old order to be executed");
+    wait_for_condition(TIMEOUT, || async {
+        let balance_after = token_a.balance_of(trader.address()).call().await.unwrap();
+        balance_before.saturating_sub(balance_after) == to_wei(10)
+            && services.get_trades(&order_id).await.unwrap().len() > 0
+    })
+    .await
+    .unwrap();
+
+    // post replacement order again
```

Comment:
given that this test now tests 2 different error conditions (order being bid on, order already filled) giving a brief explicit explanation why the error changed would be nice. I think having these small explainer comments in the test code is usually very helpful.